samples/Exporters/Console/InstrumentationWithActivitySource.cs

```diff
@@ -0,0 +1,180 @@
+                                $"{context.Request.HttpMethod}:{context.Request.Url.AbsolutePath}",
+                                ActivityKind.Server);
+
+                            string requestContent;
+                            using (var reader = new StreamReader(context.Request.InputStream, context.Request.ContentEncoding))
+                            {
+                                var headerKeys = context.Request.Headers.AllKeys;
+                                foreach (var headerKey in headerKeys)
+                                {
+                                    string headerValue = context.Request.Headers[headerKey];
+                                    activity?.AddTag($"http.header.{headerKey}", headerValue);
+                                }
+
+                                requestContent = reader.ReadToEnd();
+                                activity?.AddTag("request.content", requestContent);
+                                activity?.AddTag("request.length", requestContent.Length.ToString());
+                            }
+
+                            var echo = Encoding.UTF8.GetBytes("echo: " + requestContent);
+                            context.Response.ContentEncoding = Encoding.UTF8;
+                            context.Response.ContentLength64 = echo.Length;
+                            context.Response.OutputStream.Write(echo, 0, echo.Length);
+                            context.Response.Close();
+                        }
+                        catch (Exception)
+                        {
+                            // expected when closing the listener.
+                        }
+                    }
+                });
+            }
+
+            public void Dispose()
+            {
+                ((IDisposable)this.listener).Dispose();
+            }
+        }
+
+        private class SampleClient : IDisposable
+        {
+            private CancellationTokenSource cts;
+            private Task requestTask;
+
+            public void Start(string url)
+            {
+                this.cts = new CancellationTokenSource();
+                var cancellationToken = this.cts.Token;
+
+                this.requestTask = Task.Run(async () =>
+                    {
+                        using var source = new ActivitySource("Samples.SampleClient");
+                        using var client = new HttpClient();
+
+                        var count = 1;
+                        while (!cancellationToken.IsCancellationRequested)
+                        {
+                            var content = new StringContent($"client message: {DateTime.Now}", Encoding.UTF8);
+
+                            using (var activity = source.StartActivity("POST:" + RequestPath, ActivityKind.Client))
+                            {
+                                activity.AddBaggage("client.request.count", count.ToString(CultureInfo.InvariantCulture));
```

Comment:
did you intentionally use Baggage? Baggage is meant for properties which require to be transported to the next layer. Unless there is a reason to use it, lets not use it in the example.

Reply:
I was exploring the API and forgot to remove it.